developer/src/kmcmplib/src/NamedCodeConstants.cpp

```diff
@@ -199,17 +199,19 @@ void NamedCodeConstants::reindex()
     qsort(entries, nEntries, sizeof(NCCENTRY), sort_entries);
   }
 
-  wchar_t c = L'.', d;
+  wchar_t c = L'.', d, c_new;
   int i;
 
   for(i = 0; i < 128; i++) chrindexes[i] = -1;
 
-  if (entries != NULL) {
+  if ((entries && nEntries > 0) || !entries) {
```

Comment:
Seems pretty dangerous to execute this block if `entries` is `NULL` - we access `entries[i]`.